Source/WebCore/platform/graphics/gstreamer/eme/GStreamerEMEUtilities.cpp

```diff
@@ -29,6 +29,7 @@
 
 namespace WebCore {
 
+const char* GStreamerEMEUtilities::s_CommonEncUUID = WEBCORE_GSTREAMER_EME_UTILITIES_COMMON_ENC_UUID;
```

Comment:
There's no need to add a `#define` for `COMMON_ENC_UUID`, that's just `CLEARKEY`! You only need to change the `CLEARKEY_UUID` define here, most of the rest of the patch can then be removed.

Reply:
In case if both EME 0.1b and 3.0 versions to be support on WPE WebKit, these separate definition is required. If the old is deprecated or no need to support, the new UUID can be overwritten.

Reply:
Is there any requirement to support Clearkey for 0.1b? I would seriously doubt it :-) If not, I'd much prefer to avoid the messy code and mirror the definition as it is upstream.

Reply:
Is it not better to support both UUIDs so we can support any legacy streams?

The EME spec says "Implementations may support other SystemIDs for backwards compatibility with existing streams." So I suppose it is compliant either way.

I was just googling a bit around this subject and I notice that Android clearkey plugin supports 2 UUIDs for ClearKey https://android.googlesource.com/platform/frameworks/av/+/android-9.0.0_r3/drm/mediadrm/plugins/clearkey/common/ClearKeyUUID.cpp#25

The second one e2719d58-a985-b3c9-781a-b030af78d30e seems to be defined by Dash here https://dashif.org/identifiers/content_protection/ specifically for signalling ClearKey.

Interestingly this is different to what is currently defined as ClearKey UUID on this branch of Webkit source (58147ec8-0423-4659-92e6-f52c5ce8c3cc). I can't seem to find any hits on google for this UUID.

So I would propose adding COMMON_ENC_UUID and also updating existing ClearKey UUID to e2719d58-a985-b3c9-781a-b030af78d30e

Reply:
meant to say..we could also make this change in upstream webkit so that it stays aligned

Reply:
Thanks for digging into that @petegriffin. I still don't understand why we need this, do you have streams that fail without the "DASH" signalling UUID? I test locally with ClearKey protected content packaged for DASH and it works fine with the Common UUID in upstream.

As for the spec, I'm fine to introduce this extra complexity if there's a use-case that needs it (I still don't know what that is). However, the spec also says that  "When Initialization data is provided with initDataType "cenc", Clear Key implementations must look for and use the PSSH box with the Common SystemID.", so I'm still unclear about the motivations for this change.

Reply:
The Axinom clearkey test vectors seem to be using e2719d58-a985-b3c9-781a-b030af78d30e uuid in the mpd. See here https://github.com/Axinom/public-test-vectors. Do these streams play OK for you @charlie-ht ?

Reply:
They do not. If these streams work for you with the patch you are proposing, I won't stand in the way of getting it merged of course :-). I could only check the Audio MPD streams, I don't have decode support for `hev1.2.4.L63.90` here atm to check the videos. What I see is the demuxer struggling to get enough CENC information from the DASH stream, that may or may not be because of the (lack of) UUID, I will have to check when I have some more time to dig in. What I do see however is a ClearKey CDM being negotiated by JS, so I guess that implies it uses the modern fallback for ClearKey systems at the JS level.

Anyway, I won't be able to get this change upstream since I need tests as well :-) I'll add it to the stack of tasks and let you decide whether you need to merge it down here. Thanks for the links.